### ISSUE TITLE
cobs::decode panics on invalid data

### ISSUE THREAD

**Issue by rjsberry** [awelkie/cobs.rs#12]
Example:

```rust
use std::panic;

fn main() {
    let r = panic::catch_unwind(|| {
        let mut buf = [0; 8];
        let _ = ::cobs::decode(&[2], &mut buf);
    });

    assert!(r.is_err());
}
```

I would expect this to return `Err(())`.

It's caused by this unwrap:

https://github.com/awelkie/cobs.rs/blob/3d0d280b30b5028469eb697a2b6cfb3747f3b8f8/src/lib.rs#L337

@jamesmunns Looks like you've had an active PR here for a while with no response. If @awelkie doesn't want to continue maintaining the crate, would you consider publishing your fork?

**Comment by jamesmunns** [comment]
Hey @rjsberry, you might want to check out #14, I now have ownership of this crate.

As a temporary fix, I think https://github.com/ferrous-systems/cobs.rs/pull/2 addresses this issue (probably waaaaay too late, sorry :( ), which I have released as `postcard-cobs` v0.2.0.

I plan to re-upstream all of the work in `postcard-cobs` to the `cobs.rs` crate this weekend, hopefully.

**Comment by jamesmunns** [comment]
Tentatively closing it, as I believe it has been addressed in my `postcard-cobs` fork, which I will be merging later this year. If that DOESN'T address your problem @rjsberry, please let me know or re-open the issue!
